packages/server/core/src/base/middlewares/customServer/index.ts

```diff
@@ -168,7 +168,15 @@ export class CustomServer {
 
       const locals: Record<string, any> = {};
 
-      const customMiddlewareCtx = createCustomMiddlewaresCtx(c, locals);
+      const resArgs: ResArgs = {
+        headers: new Headers(),
+      };
+
+      const customMiddlewareCtx = createCustomMiddlewaresCtx(
+        c,
+        locals,
+        resArgs,
```

Comment:
It may not be a reasonable approach to get the value of a private variable using an external variable

Reply:
I think have no other ways to implement it.
Have any suggestions?

Reply:
Conclusion: we have some ways to modify this, but the server hook will be deprecated, so we don't modify it.